include/no-os/gpio.h

```diff
@@ -103,6 +103,20 @@ enum gpio_values {
 	GPIO_HIGH_Z
 };
 
+/**
+ * @enum gpio_pull_up
+ * @brief Enum that holds the possible pull up/ pull down resistor configuration.
+ */
+enum gpio_pull_up {
+	NO_OS_NONE,
```

Comment:
To avoid confusions, I'd go with:

NO_OS_PULL_NONE,
NO_OS_PULL_DOWN,
NO_OS_PULL_UP,
NO_OS_PULL_DOWN_WEAK,
NO_OS_PULL_UP_WEAK,